src/local-time.lisp

```diff
@@ -242,7 +242,21 @@ found."
                                             (- unix-time 86400)
                                             unix-time)
                                         transitions))
-                  (subzone (elt subzones (elt indexes transition-idx))))
+                  (subzone (elt subzones (elt indexes (max 0 transition-idx)))))
+             ;; Decide what to do when unix-time is before the first transition
+             (cond ((<= 0 transition-idx))
+                   ((and *strict-first-subzone-validity*
+                         (< (if guess-p
+                                (- unix-time (subzone-offset subzone))
+                                unix-time)
+                            (elt transitions 0)))
+                    (error "Dates before ~A are not defined in ~A"
```

Comment:
This should be a defined and documented condition, not just a default `simple-error`.